.github/workflows/ci.yml

```diff
@@ -23,6 +23,9 @@
 name: Flink Kubernetes Operator CI
 on:
   push:
+    branches:
+      - main
+      - release-*
```

Comment:
Not strictly needed for this PR, but reduces the number of build jobs to wade through (especially useful when debugging). Happy to drop/move to a new PR if desired.